### PR TITLE
Deploy api dev from slack command

### PR DESCRIPTION
**Pull request by jdgreenberger**
* Only deploy from slack commands sent by authorized users

### CODE REVIEW

File: src/deploy.coffee [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+#
+# Commands:
+#   hubot deploy dev - deploys api dev as root
+#
+# Author:
+#   Josh Greenberger
+
+exec = require('child_process').exec
+deployScriptPath = '/home/hubot/DeploymentScripts/hubot'
+authorizedSlackUsers = ['josh.greenberger', 'CircleCI']
```
Comment (jdgreenberger) [inline src/deploy.coffee]: Do we want other people to be able to deploy? Doesn't seem necessary

Comment (dan-perron) [reply src/deploy.coffee]: I dunno if there's any reason to restrict it.

File: src/deploy.coffee [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+module.exports = (robot) ->
+
+  robot.respond /deploy dev ([\w-\.]+)$/i, (msg) ->
```
Comment (dan-perron) [inline src/deploy.coffee]: Can you add a comment about what your matcher is trying to find?

File: src/deploy.coffee [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    return unless msg.message.user.name in authorizedSlackUsers
+    deployTag = msg.match[1].toLowerCase()
+    msg.send "Starting triggr_api dev continuous deploy."
+    child = exec "sudo #{deployScriptPath}/dev.deploy_api.sh #{deployTag}", (error, stdout, stderr) ->
+      if stderr
+        robot.logger.error "Error: #{stderr}"
```
Comment (dan-perron) [inline src/deploy.coffee:33]: Does robot.logger go back to slack? Would be kinda nice if it doesn't

**Review by dan-perron — APPROVED** [review]
Couple small things, otherwise LGTM!

File: src/deploy.coffee (hunk 3 above)
Comment (jdgreenberger) [reply src/deploy.coffee:33]: Nope, it goes to a log file in the hubot directory

File: src/deploy.coffee (hunk 1 above)
Comment (dan-perron) [reply src/deploy.coffee]: (restrict it to just you / circle I mean... might be nice to just add the current eng staff)

**Pull request merged**